### Error analysis

In [1]:
import os
import sys

# move to project root
os.chdir("..")

# ensure project root is importable
sys.path.append(os.getcwd())

In [2]:
from util.render_results import entityMetricsTable, spanMetricsTable

---
#### Control_fin3 model performance analysis

In [3]:
# strict 
entityMetricsTable("control_fin3", rule="strict")

,PER,LOC,ORG,O,Recall,Support
PER,6,0,16,148,0.035,170
LOC,1,16,0,18,0.457,35
ORG,3,1,10,34,0.208,48
NaNE,3,5,24,-,,
Precision,0.462,0.727,0.2,,,


In [4]:
# loose 
entityMetricsTable("control_fin3", rule="loose")

,PER,LOC,ORG,O,Recall,Support
PER,12,0,16,142,0.071,170
LOC,2,25,2,6,0.714,35
ORG,3,1,20,24,0.417,48
NaNE,3,5,24,-,,
Precision,0.6,0.806,0.323,,,


In [5]:
spanMetricsTable("control_fin3")

,Precision,Recall,F1
Metric,,,
Exact Match,0.283,0.126,0.175
Unlabeled Match,0.469,0.209,0.290
Loose Match,0.522,0.245,0.334


As we can observe, the model trained on the control dataset performs quite poorly on FIN3, suggesting training on a different domain does not achieve desirable results

---
#### fin5_only model performance analysis

In [6]:
# strict 
entityMetricsTable("fin5_only", rule="strict")

,PER,LOC,ORG,O,Recall,Support
PER,165,0,0,5,0.971,170
LOC,0,19,0,16,0.543,35
ORG,0,7,21,20,0.438,48
NaNE,14,8,34,-,,
Precision,0.922,0.559,0.382,,,


In [7]:
# loose 
entityMetricsTable("fin5_only", rule="loose")

,PER,LOC,ORG,O,Recall,Support
PER,166,0,0,4,0.976,170
LOC,0,27,2,6,0.771,35
ORG,0,7,31,10,0.646,48
NaNE,14,8,34,-,,
Precision,0.922,0.643,0.463,,,


In [9]:
spanMetricsTable("fin5_only")

,Precision,Recall,F1
Metric,,,
Exact Match,0.709,0.810,0.756
Unlabeled Match,0.734,0.838,0.782
Loose Match,0.792,0.893,0.840


Performance improves noticeably from strict exact-match evaluation (F1 = 0.756) to loose matching (F1 = 0.840), which suggests that the model often detects the correct entity region but misses the exact span boundaries. This mainly affects LOC and ORG entities, while PER predictions remain consistently strong across all evaluation settings.

---
#### synthetic model performance analysis 

In [10]:
# strict 
entityMetricsTable("synthetic", rule="strict")

,PER,LOC,ORG,O,Recall,Support
PER,150,0,0,20,0.882,170
LOC,0,15,0,20,0.429,35
ORG,2,6,5,35,0.104,48
NaNE,14,18,11,-,,
Precision,0.904,0.385,0.312,,,


In [11]:
# loose 
entityMetricsTable("synthetic", rule="loose")

,PER,LOC,ORG,O,Recall,Support
PER,160,0,0,10,0.941,170
LOC,1,24,2,8,0.686,35
ORG,2,7,9,30,0.188,48
NaNE,14,18,11,-,,
Precision,0.904,0.49,0.409,,,


In [12]:
spanMetricsTable("synthetic")

,Precision,Recall,F1
Metric,,,
Exact Match,0.685,0.672,0.679
Unlabeled Match,0.718,0.704,0.711
Loose Match,0.794,0.771,0.782


Training on the synthetic dataset alone reduces performance across all evaluation settings compared to training on FIN5, with the largest drop observed for ORG entities.

---
### mixed_6 model performance analysis

In [13]:
# strict 
entityMetricsTable("mixed_6", rule="strict")

,PER,LOC,ORG,O,Recall,Support
PER,161,0,0,9,0.947,170
LOC,0,19,0,16,0.543,35
ORG,0,1,28,19,0.583,48
NaNE,8,12,23,-,,
Precision,0.953,0.594,0.549,,,


In [15]:
# loose 
entityMetricsTable("mixed_6", rule="loose")

,PER,LOC,ORG,O,Recall,Support
PER,164,0,0,6,0.965,170
LOC,0,27,2,6,0.771,35
ORG,0,1,40,7,0.833,48
NaNE,8,12,23,-,,
Precision,0.953,0.675,0.615,,,


In [16]:
spanMetricsTable("mixed_6")

,Precision,Recall,F1
Metric,,,
Exact Match,0.751,0.822,0.785
Unlabeled Match,0.755,0.826,0.789
Loose Match,0.856,0.921,0.887


Compared to training on FIN5 only, the mixed dataset leads to a small drop in PER performance, while LOC performance remains mostly unchanged. However, ORG detection improves noticeably, especially under loose evaluation, suggesting that the synthetic data increased the model’s sensitivity to the entity class it struggled with the most.

### Conclusions

Although ORG entities are infrequent in both datasets, the poor ORG performance when training only on synthetic data suggests that the issue is not only entity frequency but also limited diversity in the generated examples. While the synthetic dataset contains a reasonable density of ORG labels, many of the generated entities are repeated names appearing in very similar contexts, which likely limits the model’s ability to generalize to unseen ORG entities in FIN3. 

Although mixing synthetic and real in-domain data improves ORG detection compared to using synthetic data alone, the overall gains remain relatively modest considering the amount of synthetic data introduced. This may be explained by limited diversity in the generated LOC and ORG examples, where many entities appear repeatedly in similar contexts. More targeted prompting could therefore improve the effectiveness of the data augmentation strategy by encouraging a wider variety of LOC and ORG entities, as well as longer and more complex spans, which remain challenging for the model.